src/Bridge/Symfony/Routing/ApiLoader.php

```diff
@@ -55,14 +59,18 @@ public function load($data, $type = null)
 
         foreach ($this->resourceNameCollectionFactory->create() as $resourceClass) {
             $resourceMetadata = $this->resourceMetadataFactory->create($resourceClass);
-            $normalizedShortName = Inflector::pluralize(Inflector::tableize($resourceMetadata->getShortName()));
+            $resourceShortName = $resourceMetadata->getShortName();
+
+            if (null === $resourceShortName) {
+                throw new InvalidResourceException(sprintf('Resource "%s" has no short name defined.', $resourceClass));
```

Comment:
@soyuka It's precisely because it won't work without a short name that we should check and not assume it's there. Just because the default implementation does the right thing, doesn't make the check unnecessary. We have discussed this before in #463 and the general agreement was that the checks are necessary... :)